.tekton/tasks/update-repository.yaml

```
@@ -0,0 +1,192 @@
+---
+# Based on: https://github.com/redhat-appstudio/build-definitions/blob/main/task/update-infra-deployments/0.1/update-infra-deployments.yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "appstudio, hacbs"
+  name: update-repository
+spec:
+  description: |
+    Clones a repository, runs script in 'SCRIPT' parameter, and generates a pull-request to the repository if a change is detected.
+  params:
+    - name: COMMIT_BRANCH
+      description: Name of the branch which that holds the update created by the Pipeline
+    - name: TARGET_BRANCH
+      description: Name of the branch which is modified by the Pipeline
+    - name: TARGET_GH_NAME
+      description: Name of the repository which is modified by the Pipeline
+    - name: TARGET_GH_OWNER
+      description: Owner of the repository which is modified by the Pipeline
+    - name: TARGET_GH_URL
+      description: URL of github repository which is modified by the Pipeline
+    - name: SCRIPT_IMAGE
+      description: Image reference to be instanciation to execute the script
```

The description is a bit unclear. I'll suggest something on the lines -
```suggestion
      description: Image reference containing a script that is executed to find updates
```

---

Description updated. The script does not have to be in the image. The image is just used to run it.